### PR TITLE
[White Label] Validate the presence of a custom tab title before creating it (+ max 20 characters length)

### LINKED ISSUE #10931: [White Label] Custom tab without a title; impossible to disable the custom tab
## Description

Follow-up from #10875.

It's possible to create a custom tab, without a title. This makes the custom tab invisible to users.

It's also not possible to disable the custom tab, even if the checkbox is disabled.

## Expected Behavior

It should not be possible to create a custom tab, without a title.
(or, a title should be mandatory, when creating a white label custom tab.)

## Actual Behaviour

It is possible to create a custom tab, without a title.
(a title is not  mandatory, when creating a white label custom tab.)

## Steps to Reproduce

As a hub

1. Visit the White label section
2. Check the checkbox: `Create custom tab in shopfront`
3. Leave the section `Title for custom tab` empty
4. Add some content to the `Content for custom tab` section
5. Click `Update`
6. Notice the Checkbox selected on point 2; it is disabled, but the empty custom section appears anyway

As a customer

1. visit the shop
2. Notice the empty (custom) section; click on it
3. Notice the custom content will be displayed

## Animated Gif/Screenshot

image

image

## Workaround

None, the user can't disable the tab. 

## Your Environment

* Version used: v4.3.11
* Browser name and version:  Firefox 113
* Operating System and version (desktop or mobile): Ubuntu 20.04

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

- Closes #10931

#### What should we test?

- As an enterprise manager, go to your enterprise white label panel
- Try to create a custom tab, but with an empty title: this should be impossible
- Try to create a custom tab, but with a title greater than 20 chars: this should be impossible

#### Release notes

Changelog Category: User facing changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/models/custom_tab_spec.rb [hunk 1]
```diff
@@ -6,4 +6,13 @@
...
     it { is_expected.to belong_to(:enterprise).required }
   end
+
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:title) }
+
+    it "should not validate title if length is greater than 20" do
+      custom_tab = build(:custom_tab, title: "a" * 21)
+      expect(custom_tab).to be_invalid
+    end
```
Comment (mkllnk) [inline spec/models/custom_tab_spec.rb:16]: https://matchers.shoulda.io/docs/v5.3.0/Shoulda/Matchers/ActiveModel.html#validate_length_of-instance_method
```suggestion
    it { is_expected.to validate_presence_of(:title) }

    it { is_expected.to validate_length_of(:title).is_at_most(20) }
```

**Review by mkllnk — APPROVED** [review]
Nice!

**Review by rioug — APPROVED** [review]
Looks good :+1:

**Comment by drummer83** [comment]
Hi @jibees,
Thanks for fixing these issues.

### Before this PR
- Creating a custom tab without title is possible. It is shown in the shopfront (if you find it) but it's shown as disabled in the white label settings.
- The title can be veeeeery long.

### After staging
- Title length is restricted to 20 characters. :heavy_check_mark: 
image
- It is not possible to create a custom tab without title. There is an error message and the values are not stored. :heavy_check_mark: 
image
- If you delete the title or create a custom tab without title, the error message is shown, but when you go to the white label settings again, the custom tab is disabled. I think it should still be enabled and show the empty title. :question: 

https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/365660f7-adf9-4d27-b9de-afeb7743ae1e

Great! I think we have great improvements here and no regressions. :partying_face: 
I will merge and create an issue for the last remaining bit.

Thanks again!

**Pull request merged**